### PR TITLE
Added support for required parameters constructors

### PR DESCRIPTION
**Pull request by brnleehng**
Proposal:
Currently, there are already static constructors that the user can use without the required parameters in the service client templates. By removing the readOnly attribute for parameters, the user can simply note the parameter to be required. This will then be added to the required parameter constructors. As of right now, setting a parameter to be readOnly and required to be true does not work.

Changes:
- Removed the readOnly extension
- Regenerated the CSharp/Azure CSharp tests
- Minor: Made doc strings in the Service Client template to be more accurate

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@brnleehng__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (brhoan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by brjohnstmsft** [comment]
Is there a related issue we can look at, preferably with examples, that explains what this change does?

**Comment by brnleehng** [comment]
The example we were looking at was.. Subscription Id is not part of the required parameter constructor. One issue is if the user doesn't set the subscription id then their operation will fail. The subscription id should just be set in the constructor if that's the case. The user shouldn't need to know that they have to set the subscription id before calling operations. This ony happens when we set the parameter to have the required attribute in the spec.

Old:
BatchManagementClient(Uri baseUri, ServiceClientCredentials credentials, params DelegatingHandler[] handlers)

New:
BatchManagementClient(Uri baseUri, ServiceClientCredentials credentials, string subscriptionId, params DelegatingHandler[] handlers)

File: AutoRest/Generators/CSharp/Azure.CSharp.Tests/Expected/AcceptanceTests/AzureBodyDuration/AutoRestDurationTestService.cs [hunk 1]
```diff
@@ -36,14 +36,14 @@ public partial class AutoRestDurationTestService : ServiceClient<AutoRestDuratio
         public Uri BaseUri { get; set; }
 
         /// <summary>
-        /// Gets or sets json serialization settings.
+        /// Gets JSON serialization settings.
```
Comment (matthchr) [inline AutoRest/Generators/CSharp/Azure.CSharp.Tests/Expected/AcceptanceTests/AzureBodyDuration/AutoRestDurationTestService.cs]: "Gets the JSON serialization settings." would be better?

File: AutoRest/Generators/CSharp/Azure.CSharp.Tests/Expected/AcceptanceTests/AzureBodyDuration/AutoRestDurationTestService.cs [hunk 2]
```diff
@@ -36,14 +36,14 @@ public partial class AutoRestDurationTestService : ServiceClient<AutoRestDuratio
...
         /// </summary>
         public JsonSerializerSettings SerializationSettings { get; private set; }
 
         /// <summary>
-        /// Gets or sets json deserialization settings.
+        /// Gets JSON deserialization settings.
         /// </summary>
-        public JsonSerializerSettings DeserializationSettings { get; private set; }        
+        public JsonSerializerSettings DeserializationSettings { get; private set; }
```
Comment (matthchr) [inline AutoRest/Generators/CSharp/Azure.CSharp.Tests/Expected/AcceptanceTests/AzureBodyDuration/AutoRestDurationTestService.cs]: Same comment here about "the"

**Comment by amarzavery** [comment]
@azuresdkci - Test this please!

**Comment by jianghaolu** [comment]
@azuresdkci retest this please

**Pull request closed without merging**